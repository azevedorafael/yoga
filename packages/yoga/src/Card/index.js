import Card from './web/Card';
import CardFooter from './web/CardFooter';
import CardHeader from './web/CardHeader';
import Plan from './web/Plan';

Card.Footer = CardFooter;
Card.Header = CardHeader;
Card.Plan = Plan;

export default Card;
