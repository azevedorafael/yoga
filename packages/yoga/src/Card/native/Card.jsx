import React from 'react';
import styled from 'styled-components';

const CardStyled = styled.View(
  ({
    theme: {
      yoga: {
        components: { card },
      },
    },
  }) => `
    padding:
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${card.backgroundColor};
    box-shadow: ${card.elevation};
    elevation: 4;
`,
);

const Card = ({ ...rest }) => <CardStyled {...rest} />;

Card.displayName = 'Card';

export default Card;
