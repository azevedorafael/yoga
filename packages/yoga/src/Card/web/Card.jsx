import React from 'react';
import styled from 'styled-components';

const CardStyled = styled.div(
  ({
    theme: {
      components: { card },
    },
  }) => `
    display: flex;
    flex-direction: column;
    justify-content: flex-start;

    width: 100%;
    padding: 
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${card.background};
    box-shadow: ${card.elevation};
`,
);

const Card = ({ ...rest }) => <CardStyled {...rest} />;

Card.displayName = 'Card';

export default Card;
