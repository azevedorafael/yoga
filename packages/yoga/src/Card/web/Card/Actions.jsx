import styled from 'styled-components';
import { arrayOf, oneOfType } from 'prop-types';

import Button from '../../../Button';
import { typeOf } from '../../../shared/propTypes';

const Actions = styled.footer``;

Actions.propTypes = {
  children: oneOfType([
    oneOfType([
      typeOf(Button),
      typeOf(Button.Text),
      typeOf(Button.Outline),
      typeOf(Button.Link),
    ]),
    arrayOf(
      oneOfType([
        typeOf(Button),
        typeOf(Button.Text),
        typeOf(Button.Outline),
        typeOf(Button.Link),
      ]),
    ),
  ]).isRequired,
};

Actions.displayName = 'Card.Actions';

export default Actions;
