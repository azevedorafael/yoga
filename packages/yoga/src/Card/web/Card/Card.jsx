import React from 'react';
import styled from 'styled-components';
import { string, shape, oneOf, node } from 'prop-types';

import Text from '../../../Text';

const CardStyled = styled.section`
  ${({
    variant,
    variantIntensity,
    theme: {
      yoga: {
        colors: { [variant]: color = {} },
        components: { card },
      },
    },
  }) => `
    padding: 
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${
      variant ? color[variantIntensity] : card.backgroundColor
    };
    box-shadow: ${card.elevation};
  `}
`;

const Ribbon = styled(Text.Tiny)`
  ${({
    variant,
    variantIntensity,
    theme: {
      yoga: {
        components: { card },
        spacing,
        colors: { gray, [variant]: color = {}, white },
      },
    },
  }) => `
    display: inline-block;
    top: ${card.padding.top}px;
    left: 0;

    margin-left: -${card.padding.left}px;
    margin-bottom: ${card.padding.bottom}px;

    padding: 
      ${spacing.xxsmall}px
      ${spacing.small}px
      ${spacing.xxsmall}px
      ${spacing.medium}px;
    
    border-top-right-radius: ${card.ribbon.radius}px;
    border-bottom-right-radius: ${card.ribbon.radius}px;

    background-color: ${
      variant
        ? color[typeof variantIntensity === 'number' ? variantIntensity : 2]
        : gray.i10
    };

    color: ${variant ? white : gray.i80};
  `}
`;

const Card = ({ ribbon, children, ...rest }) => (
  <CardStyled {...rest}>
    {Object.keys(ribbon).length > 0 && (
      <Ribbon
        variant={ribbon.variant}
        variantIntensity={ribbon.variantIntensity}
        as="span"
      >
        {ribbon.text}
      </Ribbon>
    )}
    {children}
  </CardStyled>
);

Card.propTypes = {
  /** text: the content inside the Card Ribbon
   * variant: style the ribbon following the theme (primary, secondary, tertiary)
   * variantIntensity: ribbon variant color intensity (0, 1, 2) */
  ribbon: shape({
    text: string,
    variant: oneOf(['', 'primary', 'secondary']),
    variantIntensity: oneOf([0, 1, 2]),
  }),
  children: node,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['', 'primary', 'secondary']),
  /** intensity of variant color (0, 1, 2) */
  variantIntensity: oneOf([0, 1, 2]),
};

Card.defaultProps = {
  ribbon: {},
  children: null,
  variant: '',
  variantIntensity: 2,
};

Card.displayName = 'Card';

export default Card;
