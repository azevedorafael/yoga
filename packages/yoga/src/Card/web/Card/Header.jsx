import styled from 'styled-components';

const Header = styled.header``;

Header.displayName = 'Card.Header';

export default Header;
