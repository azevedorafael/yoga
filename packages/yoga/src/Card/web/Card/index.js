import Card from './Card';
import Header from './Header';
import Content from './Content';
import Actions from './Actions';

Card.Header = Header;
Card.Content = Content;
Card.Actions = Actions;

export default Card;
