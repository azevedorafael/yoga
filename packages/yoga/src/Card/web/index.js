import Card from './Card';
import Footer from './Footer';
import Header from './Header';
import Plan from './Plan';

export { Card, Footer, Header, Plan };
