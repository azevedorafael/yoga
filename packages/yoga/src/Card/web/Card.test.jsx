import React from 'react';
import { render } from '@testing-library/react';
import ThemeProvider from '../../ThemeProvider';
import Card from './Card';

describe('<Card />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default Card', () => {
      const { container } = render(
        <ThemeProvider>
          <Card>
            <h1>Hello World</h1>
          </Card>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
