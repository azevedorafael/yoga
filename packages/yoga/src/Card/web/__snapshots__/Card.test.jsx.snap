// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot Plan Card with secondary prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  padding: 16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 .content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c1 h1,
.c1 h2,
.c1 span {
  margin: 0;
}

.c3 {
  font-size: 12px;
  color: #6B6B78;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  color: undefined;
}

.c5 {
  font-size: 12px;
  padding-top: 4px;
  color: #6B6B78;
}

.c7 {
  font-size: 14px;
  font-weight: 700;
  color: #178173;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
      style="margin-bottom: 40px;"
    >
      <div
        class="content"
      >
        <h1
          class="c3"
        >
          Basic
        </h1>
      </div>
    </div>
    <h2
      class="c4"
    >
      U$ 19.90
    </h2>
    <span
      class="c5"
    >
      /
      month
    </span>
    <div
      class="c6"
      style="margin-top: 20px;"
    >
      <span
        class="c7"
      >
        1,239
         gyms
      </span>
    </div>
  </div>
</div>
`;

exports[`<Card /> should match snapshot with default Card 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  padding: 16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

<div>
  <div
    class="c0"
  >
    <h1>
      ol√° mundo
    </h1>
  </div>
</div>
`;
